add_base_remote.sh script should not fail if run in non-git environment

When app is deployed with no git environment (i.e. no clone, just clean checkout), the script is being run after npm install and fails.

```
[Parsing package.json]
[INFO] Parsing package.json
[INFO] No file to start server
[INFO] either use 'docker run' to start container or add index.js or server.js
[INFO] Using npm v8.1.4 from Node
[INFO] Installing node modules from ./package-lock.json

> budapest-api@0.1.0 postinstall
> sh ./scripts/add_base_remote.sh

fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
npm notice 
npm notice New minor version of npm available! 8.1.4 -> 8.2.0
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v8.2.0>;
npm notice Run `npm install -g npm@8.2.0` to update!
npm notice 
npm ERR! code 128
npm ERR! path /workspace
npm ERR! command failed
npm ERR! command sh -c sh ./scripts/add_base_remote.sh
```

In such cases, it should skip instead of hard failure.

--- a/scripts/add_base_remote.sh
+++ b/scripts/add_base_remote.sh
@@ -1,9 +1,26 @@
 # Создаёт git remote на базовый репозиторий, если его (remote) ещё нет
 
-export BASE_REMOTE_URL=git@github.com:gooditcollective/base.git
+BASE_REMOTE_URL=git@github.com:gooditcollective/base.git
 
-# если мы НЕ в CI (github actions, vercel)
-if [ -z "$CI" ]; then
-  # создаёт remote, если его нет: https://stackoverflow.com/a/57936187
-  git config remote.base.url >&- || git remote add base $BASE_REMOTE_URL
+if [ -n "$CI" ]; then
+  echo "We in CI/CD, skipping"
+  exit 0
 fi
+
+if ! [ -x "$(command -v git)" ]; then
+  echo "git command not found, skipping"
+  exit 0
+fi
+
+if ! [ "$(git rev-parse --is-inside-work-tree 2>/dev/null)" ]; then
+  echo "Not a git repository, skipping"
+  exit 0
+fi
+
+if git config remote.base.url; then
+  echo "Base remote already set, skipping"
+  exit 0
+fi
+
+git remote add base $BASE_REMOTE_URL
+echo "Base remote added"

